app/controllers/auth/omniauth_callbacks_controller.rb

```diff
@@ -97,11 +99,28 @@ def github_member?(conf)
   end
 
   # Get user's teams and check if one match to restriction.
-  def member_of(url)
+  def member_of(url, per_page: nil)
     # Get user's groups.
     token = request.env["omniauth.auth"].credentials["token"]
-    resp = Faraday.get url, access_token: token
-    teams = JSON.parse resp.body
+    teams = []
+    # groups are paginated !
+    np = 0
+    loop do
+      np += 1
+      resp = Faraday.get url, { page: np, per_page: per_page,
+                                access_token: token }.compact
+      teams.concat JSON.parse resp.body
+      # -> gitlab: x-next-page is in headers
+      # -> github: Link is in headers
+      #            and if we are not on last page, we have a last link
+      gitlab_next = resp.headers.key?("x-next-page") && \
+                    resp.headers["x-next-page"].present?
+      github_next = resp.headers.key?("Link") && \
+                    resp.headers["Link"].include?('rel="last"') && \
+                    !resp.headers.key?("x-next-page")
```

Comment:
Nitpicking: Maybe you could move these `gitlab_next` and `github_next` variables intro separate auxiliar methods, since they make the loop larger than it should.